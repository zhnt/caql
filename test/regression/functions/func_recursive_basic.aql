function factorial(n) {
    if (n <= 1) {
        return 1
    }
    return n * factorial(n - 1)
}

function fibonacci(n) {
    if (n <= 1) {
        return n
    }
    return fibonacci(n - 1) + fibonacci(n - 2)
}

function power(base, exp) {
    if (exp == 0) {
        return 1
    }
    if (exp == 1) {
        return base
    }
    return base * power(base, exp - 1)
}

print("factorial(5) = " + string(factorial(5)))
print("fibonacci(6) = " + string(fibonacci(6)))
print("power(2, 4) = " + string(power(2, 4)))
print("power(3, 3) = " + string(power(3, 3)))
