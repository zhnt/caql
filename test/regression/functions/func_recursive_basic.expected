factorial(5) = 120
fibonacci(6) = 8
power(2, 4) = 16
power(3, 3) = 27


